app/views/contexts/summary/list_collector_block.py

```diff
@@ -107,6 +107,9 @@ def list_summary_element(self, summary: Mapping[str, Any]) -> dict[str, Any]:
         list_summary_context = self.list_context(
             list_collector_block["summary"],
             for_list=list_collector_block["for_list"],
+            list_collector_location=Location(section_id=self._location.section_id,
```

Comment:
It looks like this passes various attributes of the list collector block to ListContext, which calls the build_list_items_context which then looks up the block again - could you just pass down the list collector block instead?

Reply:
Brilliant point, will have a go at amending to that!

Reply:
I don't mind either approach, ideally a function should be given just what it needs.
- If you pass the block, then you can remove a bunch of these args
- Otherwise, pass in whether it has repeating and a section id?.